### PR TITLE
026 implement table resource

### LINKED ISSUE #26: Implement `Table` resource
https://docs.snowflake.net/manuals/sql-reference/sql/create-table.html

```python
table = Table(
    resource_name='acme-dev-tracking',
    name='tracking',
    database=db,
    schema=schema,
    comment='Tracking data from the website',
    cluster_by=['to_date(timestamp)', 'id'],
    data_retention_time_in_days=7,
)
```

### PR DESCRIPTION
**Pull request by cloudspeak**
Implemented the Table resource and dynamic provider, allowing the creation of Snowflake tables in Pulumi. For example:

```python
my_table = Table("MyTable",
                 database=my_database.name,
                 schema=my_schema.name,
                 columns=[
                     Column("col_1", "INT", primary_key=True, autoincrement=True),
                     Column("col_2", "VARCHAR", unique=True, collation="utf8"),
                     Column("col_3", "INT", not_null=True, default="123"),
                 ],
                 comment="A test table",
                 data_retention_time_in_days=1
                 )
```

I used my judgement to decide which column attributes belong in the "advanced" milestone, so let me know if you agree. Currently the columns support:
* PRIMARY KEY
* UNIQUE
* DEFAULT
* AUTOINCREMENT
* COLLATE

Closes #26

### CODE REVIEW

File: example/__main__.py [hunk 1]
```diff
@@ -7,23 +7,59 @@
...
+# you will need:
+#
+#   * Snowflake credentials in your config (see README)
+#   * a database named "MyDatabase2" and schema named "MySchema2" (or comment out the "Explicit Provider" example below)
+
+
+# DEFAULT PROVIDER EXAMPLE
+
+# The resources below all use the default provider, which will read the credentials from config.  We can also
+# specify the database and schema in the config, but in this case they are passed in explicitly.
```
Comment (LouisAmon) [inline example/__main__.py]: Could you please refactor to multiline docstring in markdown syntax ?

File: example/__main__.py [hunk 2]
```diff
@@ -38,34 +74,21 @@
...
-my_schema = Schema("MySchema",
-                   database=my_database.name,
-                   comment="A test schema",
-                   transient=True,
-                   data_retention_time_in_days=1
-                   )
+my_pipe = Pipe("MyPipe",
+               auto_ingest=False,
+               comment="A test pipe",
+               code=get_pipe_inputs().apply(lambda ins: f"copy into {ins['table_name']} from @{ins['stage_name']}"),
```
Comment (LouisAmon) [inline example/__main__.py]: Double `apply`... seems a bit circumvoluted doesn't it ?

Isn't there a simpler way ?

Comment (LouisAmon) [reply example/__main__.py]: Maybe just:

```python
code=pulumi.Output.all(my_table.full_name, my_stage.full_name).apply(
    lambda a: f"COPY INTO {a[0]} FROM @{a[1]}"
)
```

File: example/__main__.py [hunk 3]
```diff
@@ -77,32 +100,36 @@
...
                          )
 
-# Pipe example - this required a table named FIRSTTABLE to exist.
 
-my_pipe = Pipe("MyPipe",
-               auto_ingest=False,
-               comment="A test pipe",
-               code=my_stage.name.apply(lambda n: f"copy into FIRSTTABLE from @{n}")
-               )
+# EXPLICIT PROVIDER EXAMPLE
```
Comment (LouisAmon) [inline example/__main__.py]: maybe separate into its own file for the second example ?

```
examples/simple.py
examples/explicit_provider.py
```

?

Comment (cloudspeak) [reply example/__main__.py]: I'd like to, but Pulumi only supports `__main__.py` so I'd have to create a new folder with its own version of the config, requirements.txt, etc.  That just felt a bit overkill when I was only demonstrating different ways of instantiating resources.  What do you think?

Comment (LouisAmon) [reply example/__main__.py]: Doesn't really matter though ?

They are just examples of what a `__main__.py` would look like :)

It's just an aesthetic comment anyway, PR is approved so I leave it up to you what you prefer for this matter

Comment (cloudspeak) [reply example/__main__.py]: True!  I think I'll keep it as it is this time - I think there is value in having an example which doesn't need code changes and can be deployed with a couple of commands :)

**Pull request merged**